lib/internal/modules/esm/hooks.js

```diff
@@ -127,28 +127,31 @@ class Hooks {
    * Import and register custom/user-defined module loader hook(s).
    * @param {string} urlOrSpecifier
    * @param {string} parentURL
+   * @param {any} data
    */
-  async register(urlOrSpecifier, parentURL) {
+  async register(urlOrSpecifier, parentURL, data) {
     const moduleLoader = require('internal/process/esm_loader').esmLoader;
     const keyedExports = await moduleLoader.import(
       urlOrSpecifier,
       parentURL,
       kEmptyObject,
     );
-    this.addCustomLoader(urlOrSpecifier, keyedExports);
+    return await this.addCustomLoader(urlOrSpecifier, keyedExports, data);
```

Comment:
```suggestion
    return this.addCustomLoader(urlOrSpecifier, keyedExports, data);
```

Reply:
Is there a reason it's preferred to avoid `await`ing async calls in async functions if they are the terminal call?

Reply:
it's redundant and adds an artificially forced tick, thus making it slower.

Reply:
TIL. Thanks!